### PR TITLE
Feature card ui component

### PR DESCRIPTION
**Pull request by ddave592**
init card UI

to-do:
- [x] title
- [x] image
- [x] address
- [x] price
- [x] tags
- [x] description
- [x] ratings
- [x] wait time
- [x] dataset

additional:
- [x] randomly generate card from json file

ready for review

continuation from PR #2

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import React from 'react';
+import CardUi from './components/CardUi'
+import NavBar from './components/navbar';
```
Comment (bonno42h) [inline src/App.js]: gotta setup webpack or jsconfig file to do absolute imports. if u want, i will do it myself and update imports once this is pushed, let me know

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,33 @@
 import React from 'react';
+import CardUi from './components/CardUi'
```
Comment (bonno42h) [inline src/App.js:2]: Should make the imports absolute to src folder via webpack or jsconfig. Will do that and update imports once the PR is merged.

File: src/App.js [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
+import NavBar from './components/Navbar';
 import './styles/App.scss';
+import restaurants from './data/restaurants.json';
+
+function deepCopy (data) {
+  return JSON.parse(JSON.stringify(data))
```
Comment (bonno42h) [inline src/App.js]: semicolon missing

File: src/App.js [hunk 4]
```diff
@@ -1,12 +1,33 @@
 import React from 'react';
+import CardUi from './components/CardUi'
+import NavBar from './components/Navbar';
 import './styles/App.scss';
+import restaurants from './data/restaurants.json';
+
+function deepCopy (data) {
```
Comment (bonno42h) [inline src/App.js]: Unexpected space before function parentheses. eslint(space-before-function-paren)

File: src/App.js [hunk 5]
```diff
@@ -1,12 +1,33 @@
...
+  return JSON.parse(JSON.stringify(data))
+}
+
+function ShuffleArr (input) {
```
Comment (bonno42h) [inline src/App.js]: Unexpected space before function parentheses.eslint(space-before-function-paren)

File: src/App.js [hunk 6]
```diff
@@ -1,12 +1,33 @@
...
+    let randomIndex = Math.floor(Math.random() * (i + 1));
+    let itemAtIndex = input[randomIndex];
+
+    input[randomIndex] = input[i];
+    input[i] = itemAtIndex;
+  }
+  return input;
+}
+
+let cardCycle = ShuffleArr(deepCopy(restaurants));
```
Comment (bonno42h) [inline src/App.js]: if it does not change - do a `const `instead of `let`

File: src/components/CardUi.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            <h1>
+              {capFirstLetter(list[cardIndex].name)}
+            </h1>
+            <div className="break" />
+            <div className="short-info">
+                <RoomIcon />
+                <span>
+                  {list[cardIndex].location}
+                </span>
+              <Chip label={'$'.repeat(list[cardIndex].price)} className="card-cost" color="secondary" size="medium" variant="outlined" />
```
Comment (bonno42h) [inline src/components/CardUi.js]: bit long of a line. perhaps break into seperate ones?

**Review by bonno42h — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 3 above)
Comment (ddave592) [reply src/App.js]: fixed 22746e9af0c79aaa35981b436f9fb4623e060e38

File: src/App.js (hunk 4 above)
Comment (ddave592) [reply src/App.js]: fixed 22746e9af0c79aaa35981b436f9fb4623e060e38

File: src/App.js (hunk 6 above)
Comment (ddave592) [reply src/App.js]: fixed 22746e9af0c79aaa35981b436f9fb4623e060e38

File: src/components/CardUi.js (hunk 7 above)
Comment (ddave592) [reply src/components/CardUi.js]: fixed 22746e9af0c79aaa35981b436f9fb4623e060e38

File: src/App.js (hunk 1 above)
Comment (ddave592) [reply src/App.js]: Yeah that would be great if you can, I would do it myself but I've not done absolute imports through configs before.

Thanks,

Comment (ddave592) [reply src/App.js]: @bonno42 are we making a pull request for this? let me know, meanwhile I'll resolve these comments so we can look to merge this PR as soon as we can

Comment (bonno42h) [reply src/App.js]: I will make a separate PR once this one is merged.

**Review by bonno42h — APPROVED** [review]

**Pull request merged**